docs: add guides so users can get started faster

R44VC0RP opened the pull request:
## Summary

- Adds a new **Guides** section to the documentation navigation
- Creates a comprehensive "First Hour" guide to help new users get productive quickly

## What's in the guide

The guide follows a progressive disclosure pattern - essentials up front, details as you read further:

1. **Quick start essentials** - bullet list of the 5 things you need to know
2. **Installation** - links to install options
3. **Provider setup** - promotes OpenCode Zen, includes `opencode auth login` option
4. **Project initialization** - `/init` command
5. **File references** - using `@` to reference files
6. **Plan-Build workflow** - core workflow explanation with table
7. **Undo/redo** - safety net for mistakes
8. **Shell commands** - `!` prefix
9. **Sharing** - `/share` command
10. **Quick tips** - context, mentoring tone, image paste (Ctrl+V)
11. **MCP servers** - Context7 and Exa setup with:
    - Config location explanation (global vs project)
    - Global config example for Context7/Exa
    - Project config example for Neon Postgres
12. **Cheat sheet** - quick reference table

## Changes

- `packages/web/astro.config.mjs` - Added Guides section to sidebar
- `packages/web/src/content/docs/guides/first-hour.mdx` - New guide (356 lines)

R44VC0RP committed:
docs: add First Hour guide for new users

- Create new Guides section in docs navigation
- Add comprehensive first-hour.mdx guide covering:
  - Installation and provider setup (promoting Zen)
  - Project initialization with /init
  - File references with @
  - Plan-Build workflow explanation
  - Undo/redo functionality
  - Shell command integration
  - Image paste support (Ctrl+V)
  - MCP server setup (Context7, Exa) with config location guidance
  - Quick tips and cheat sheet
- Structure guide as progressive disclosure (essentials first, details later)

R44VC0RP commented:
OC Session: https://opncd.ai/share/bI5W0WPS

R44VC0RP committed:
docs: add Subagents guide

- Add comprehensive guide for working with subagents
- Cover built-in subagents (@explore, @general)
- Document manual invocation with @ mentions
- Explain automatic delegation and session navigation
- Full configuration reference with examples:
  - description, mode, model, temperature, top_p
  - prompt, tools, permission, maxSteps, color, disable
- Include 5 practical example subagents:
  - Code reviewer, Test writer, Documentation writer
  - Security auditor, Git committer
- Add best practices and cheat sheet
- Update sidebar to include subagents guide

R44VC0RP committed:
Merge branch 'dev' into docs-first-hour-guide

rekram1-node commented:
Going to have Jay look over this

R44VC0RP commented:
<img width="684" height="356" alt="image" src="https://github.com/user-attachments/assets/43cec910-cea7-4d48-872b-4ca975549994" />

already helped 2 people

R44VC0RP committed:
docs: add Commands guide for automation workflows

- Add comprehensive guide for custom commands
- Cover template syntax: $ARGUMENTS, $1/$2, shell injection, file refs
- Document all configuration options
- Include 7 practical example commands:
  - Test runner, PR reviewer, Commit helper
  - Documentation generator, Dependency checker
  - Quick refactor, Type checker
- Explain subtask mode and subagent integration
- Add best practices and cheat sheet
- Update sidebar to include commands guide

jayair commented:
I think the first hour one should be merged into this doc, they are both doing very similar things, maybe rename section in the current doc to emphasize that it's a guide for new users: https://opencode.ai/docs/#initialize

The commands doc has similar content to this one, I think we should merge that too: https://opencode.ai/docs/commands/ 

We cover subagents here: https://opencode.ai/docs/agents/#subagents. We should probably add to that doc if we can, but open to creating a new doc for this.

R44VC0RP commented:
@jayair yeah good points

heres my thinking for adding the guides: 

the existing docs are reference-oriented - they document all options and configurations. Great for "how do I configure X?" but less helpful for "when should I use X in my workflow?"

the guides add a TL;DR section at the top (learn/understand/value in 30 seconds)

for example, commands.mdx documents all 5 config options. The guide shows 8 real-world commands (PR reviewer, test runner, etc.) and explains when to use subtask: true.

thoughts?

R44VC0RP commented:
I think I will merge the commands guide with the commands docs reference, after reviewing more it looks like it fits better.

R44VC0RP committed:
docs: merge commands guide into reference doc

- Merge guides/commands.mdx content into commands.mdx
- Add 'essentials' summary at top
- Add quick example section
- Add practical example commands (test runner, PR reviewer, etc.)
- Add subagent integration section
- Add cheat sheet
- Remove separate guides/commands.mdx file
- Update sidebar navigation
